client/src/component/FlightSearchComponent/FlightSearch.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useRef } from 'react';
+
+import { Grid, Paper, TextField, Button, InputLabel } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+
+import useStyles from './styles';
+
+// Mock Data
+const cities = [
+  { title: 'Vancouver' },
+  { title: 'Calgary' },
+  { title: 'Toronto' },
+  { title: 'Bangkok' },
+];
+
+const FlightSearch = () => {
+  const classes = useStyles();
+  const ref0 = useRef();
+  const curr = new Date();
+  curr.setDate(curr.getDate());
+  const date = curr.toISOString().substr(0, 10);
+
+  const [state, setState] = useState({
+    from: 'Vancouver',
+    to: 'Bangkok',
+    arrival: date,
+    departure: date,
+    travellers: 1,
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setState((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const onSelect = (e, v, r) => {
+    const name = ref0.current.getAttribute('name');
+    setState({ ...state, [name]: v });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  return (
+    <div className={classes.root}>
+      <form onSubmit={handleSubmit}>
+        <Paper className={classes.paperContainer} elevation={7}>
+          <Grid className={classes.input} lg={2} sm={3} xs={6} item>
+            <InputLabel>From</InputLabel>
+            <Autocomplete
+              freeSolo
+              id='from'
+              ref={ref0}
+              name='from'
+              options={cities.map((city) => city.title)}
+              defaultValue={state.from}
+              value={state.from}
+              onChange={onSelect}
```

Comment:
I notice here and below, that when you're calling `onSelect` you're not passing it the `v` or `r` parameters the function calls for. With the function as it stands, it will assign the value of that particular property of the state to `undefined`.